### PR TITLE
add support for Python 3.9

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
## Description
Adds Python 3.9 support by updating the setup.py as well as our test suite.

It went fairly smoothly other than the versions of gevent that we test with not supporting python 3.9.

We also don't test any recent gevents which we should look into doing.

## Checklist
- [x] Entry added to release notes using [reno](https://docs.openstack.org/reno/latest/user/usage.html)
- [x] Tests provided; and/or
- [ ] ~Description of manual testing performed and explanation is included in the code and/or PR.~
- [x] ~Library documentation is updated.~
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

**Review by jd — DISMISSED** [review]

**Comment by jd** [comment]
Add a reno entry?

File: ddtrace/contrib/asyncio/compat.py [hunk 1]
```diff
@@ -1,10 +1,11 @@
...
 
-    asyncio_current_task = asyncio.Task.current_task
+def asyncio_current_task():
+    if hasattr(asyncio, "current_task"):
+        try:
+            return asyncio.current_task()
+        except RuntimeError:
+            return None
+    else:
+        return asyncio.Task.current_task()
```
Comment (jd) [inline ddtrace/contrib/asyncio/compat.py:11]: Do this once and for all like it was before it a little more efficient that doing `if` on each call.

File: tests/contrib/asyncio/test_tracer.py [hunk 2]
```diff
@@ -16,7 +17,10 @@ class TestAsyncioTracer(AsyncioTestCase):
     def test_get_call_context(self):
         # it should return the context attached to the current Task
         # or create a new one
-        task = asyncio.Task.current_task()
+        if hasattr(asyncio.Task, "current_task"):
```
Comment (jd) [inline tests/contrib/asyncio/test_tracer.py:20]: use `ddtrace.contrib.asyncio.compat`?

**Review by jd — CHANGES_REQUESTED** [review]

File: ddtrace/contrib/asyncio/compat.py (hunk 1 above)
Comment (Kyle-Verhoog) [reply ddtrace/contrib/asyncio/compat.py:11]: Ah yes, of course, good catch

**Review by jd — APPROVED** [review]

**Pull request merged**